### PR TITLE
Add support for decoding tiff's with YCbCr PhotometricInterpretation

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [ ] I have provided test coverage for my change (where applicable) **note: i could not provide test images for planar ycbcr with subsampling > 1. No tool I know of, could write those.**

### Description

Add support for decoding YCbCr tiff images with contiguous and planar pixel layout.

### CODE REVIEW

File: src/ImageSharp/Formats/Tiff/PhotometricInterpretation/YCbCrConverter.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            Rgba32 rgba = this.converter.Convert(yExpanded, cbExpanded, crExpanded);
+
+            return rgba;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static byte RoundAndClampTo8Bit(float value)
+        {
+            int input = (int)MathF.Round(value);
+            return (byte)Math.Min(Math.Max(input, 0), 255);
```
Comment (gfoidl) [inline src/ImageSharp/Formats/Tiff/PhotometricInterpretation/YCbCrConverter.cs:71]:
```suggestion
            return (byte)Math.Clamp(input, 0, 255);
```
? (and if it's available for all TFMs).

Right now codegen is pretty much equal, but in a future JIT there could be some improvements in that area.

**Review by gfoidl — COMMENTED** [review]
A "nit-suggestion".

File: src/ImageSharp/Formats/Tiff/PhotometricInterpretation/YCbCrConverter.cs (hunk 1 above)
Comment (brianpopow) [reply src/ImageSharp/Formats/Tiff/PhotometricInterpretation/YCbCrConverter.cs]: yes makes sense, thanks

Comment (brianpopow) [reply src/ImageSharp/Formats/Tiff/PhotometricInterpretation/YCbCrConverter.cs]: unfortunately not available with `net472`, using `Numerics.Clamp` now.

Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Tiff/PhotometricInterpretation/YCbCrConverter.cs]: Numerics version is faster than the runtime one I think

Comment (gfoidl) [reply src/ImageSharp/Formats/Tiff/PhotometricInterpretation/YCbCrConverter.cs]: > Numerics version is faster than the runtime one I think

The runtime one has an additional check for min > max which isn't needed here as the limits are constants (0 and 255). After inlineing the JIT is able to remove that min > max check, so codegen is quite similar. The difference is that the Numerics-version checks value > max first, while the runtime-version checks value < min first. So which version is faster depends on the likeness of value < min vs value > max and the branch-predictor's history (and potentially on dynamic profile guided optimization).

`Numerics.Clamp` is good to use here. Maybe just add a `Debug.Assert(min < max)` there to catch "typos" early?

**Review by JimBobSquarePants — APPROVED** [review]
LGTM.

Something to look at in the future would be vectorizing the planar conversion. I'm confident that we can do some optimization there.

**Pull request merged**